registrations/tasks.py

```diff
@@ -435,8 +440,66 @@ def get_authority_from_source(self, source):
             'OPTOUT USSD App': 'optout',
             'CLINIC USSD App': 'clinic',
             'CHW USSD App': 'chw',
+            'NURSE USSD App': 'nurse',
         }.get(source.name)
 
+    def run(self, registration_id, **kwargs):
+        self.l.info("Compiling Jembi Json data")
+        from .models import Registration
+        registration = Registration.objects.get(pk=registration_id)
+        authority = self.get_authority_from_source(registration.source)
+        if authority is None:
+            self.l.error(
+                'Unable to establish authority for source %s. Skipping.' % (
+                    registration.source))
+            return
+
+        json_doc = self.build_jembi_json(registration)
+        try:
+            result = requests.post(
+                self.URL,
+                headers={'Content-Type': 'application/json'},
+                data=json.dumps(json_doc),
+                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
+                verify=False
+            )
+            result.raise_for_status()
+            return result.text
+        except (HTTPError,) as e:
+            # retry message sending if in 500 range (3 default retries)
+            if 500 < e.response.status_code < 599:
+                raise self.retry(exc=e)
+            else:
+                self.l.error('Error when posting to Jembi. Payload: %r' % (
+                    json_doc))
+                raise e
+        except (Exception,) as e:
+            self.l.error(
+                'Problem posting Registration %s JSON to Jembi' % (
+                    registration_id), exc_info=True)
+
+
+class PushRegistrationToJembi(BasePushRegistrationToJembi, Task):
+    """ Task to push registration data to Jembi
+    """
+    name = "ndoh_hub.registrations.tasks.push_registration_to_jembi"
+    l = get_task_logger(__name__)
+    URL = "%s/subscription" % settings.JEMBI_BASE_URL
+
+    def get_subscription_type(self, authority):
```

Comment:
I think the diff is a little strange, it looks like I renamed `PushRegistrationToJembi` to `BasePushRegistrationToJembi` and moved functions out to a new `PushRegistrationToJembi` class; when instead I made a new `BasePushRegistrationToJembi` and moved some functions out from `PushRegistrationToJembi` to it, so you kind of have to read the diff in reverse.

This function is still in the `PushRegistrationToJembi` class, and called from the `build_jembi_json` function.